librz/core/cmd_analysis.c

```diff
@@ -9251,3 +9234,734 @@ RZ_IPI int rz_cmd_analysis(void *data, const char *input) {
+}
+
+RZ_IPI RzCmdStatus rz_analysis_function_vars_rename_handler(RzCore *core, int argc, const char **argv) {
+	const char *newname = argv[1];
+	const char *oldname = argv[2];
+	bool result = rz_core_analysis_var_rename(core, oldname, newname);
+	return result ? RZ_CMD_STATUS_OK : RZ_CMD_STATUS_OK;
+}
+
+static RzCmdStatus analysis_function_vars_accesses(RzCore *core, int access_type, const char *varname) {
+	RzAnalysisFunction *fcn = rz_analysis_get_fcn_in(core->analysis, core->offset, -1);
+	if (!fcn) {
+		eprintf("Cannot find function in 0x%08" PFMT64x "\n", core->offset);
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	if (!varname) {
+		RzList *list = rz_analysis_var_all_list(core->analysis, fcn);
+		RzListIter *iter;
+		RzAnalysisVar *var;
+		rz_list_foreach (list, iter, var) {
+			var_accesses_list(fcn, var, NULL, access_type, var->name);
+		}
+		rz_list_free(list);
+	} else {
+		RzAnalysisVar *var = rz_analysis_function_get_var_byname(fcn, varname);
+		if (!var) {
+			eprintf("Cannot find variable %s\n", varname);
+			return RZ_CMD_STATUS_ERROR;
+		}
+		var_accesses_list(fcn, var, NULL, access_type, var->name);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_function_vars_reads_handler(RzCore *core, int argc, const char **argv) {
+	return analysis_function_vars_accesses(core, RZ_ANALYSIS_VAR_ACCESS_TYPE_READ, argv[1]);
+}
+
+RZ_IPI RzCmdStatus rz_analysis_function_vars_writes_handler(RzCore *core, int argc, const char **argv) {
+	return analysis_function_vars_accesses(core, RZ_ANALYSIS_VAR_ACCESS_TYPE_WRITE, argv[1]);
+}
+
+RZ_IPI RzCmdStatus rz_analysis_function_vars_type_handler(RzCore *core, int argc, const char **argv) {
+	RzAnalysisFunction *fcn = rz_analysis_get_fcn_in(core->analysis, core->offset, -1);
+	if (!fcn) {
+		eprintf("Cannot find function in 0x%08" PFMT64x "\n", core->offset);
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	RzAnalysisVar *v = rz_analysis_function_get_var_byname(fcn, argv[1]);
+	if (!v) {
+		eprintf("Cannot find variable %s\n", argv[1]);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_analysis_var_set_type(v, argv[2]);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_function_vars_xrefs_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	RzAnalysisFunction *fcn = rz_analysis_get_fcn_in(core->analysis, core->offset, -1);
+	if (!fcn) {
+		eprintf("Cannot find function in 0x%08" PFMT64x "\n", core->offset);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	PJ *pj = NULL;
+	if (mode == RZ_OUTPUT_MODE_JSON) {
+		pj = pj_new();
+		pj_o(pj);
+		pj_k(pj, "reads");
+	} else {
+		rz_cons_printf("afvR\n");
+	}
+	list_vars(core, fcn, pj, 'R', argv[1]);
+	if (mode == RZ_OUTPUT_MODE_JSON) {
+		pj_k(pj, "writes");
+	} else {
+		rz_cons_printf("afvW\n");
+	}
+	list_vars(core, fcn, pj, 'W', argv[1]);
+	if (mode == RZ_OUTPUT_MODE_JSON) {
+		pj_end(pj);
+		char *j = pj_drain(pj);
+		rz_cons_printf("%s\n", j);
+		free(j);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+static int outputmode2type(RzOutputMode mode) {
```

Comment:
This maybe to put somewhere in `librz/core/core.c` or similar, you will need it in other command handlers too.